index.html

```diff
@@ -604,6 +616,149 @@ <h3>Set the session description</h3>
+        </p>
+      </li>
+      <li>
+        <p>
+          Let |accepted:boolean| be the result of [= fire an event | firing an event =] named
+          {{RTCIceTransport/candidatepairpruneproposed}} at |transport|, using {{RTCIceCandidatePairEvent}}, with the
+          {{Event/cancelable}} attribute initialized to <code>true</code>, and the {{RTCIceCandidatePairEvent/candidatePair}}
+          attribute initialized to |candidatePair|.
+        </p>
+      </li>
+      <li>
+        <p>
+          If |accepted| is <code>false</code>, instruct the [= ICE agent =] to not prune the candidate pair indicated by |candidatePair|, and instead continue to send and respond to ICE connectivity checks on the candidate pair as before.
+        </p>
+      </li>
+      <li>
+        <p>
+          Otherwise, instruct the [= ICE agent =] to prune the candidate pair indicated by |candidatePair|.
+        </p>
+      </li>
+    </ol>
+    <pre class="idl">
+      partial interface RTCIceTransport {
+        attribute EventHandler oncandidatepairadded;
+        attribute EventHandler oncandidatepairpruneproposed;
+      };</pre>
+    <section>
+      <h2>Attributes</h2>
+      <dl data-link-for="RTCIceTransport" data-dfn-for="RTCIceTransport" class="attributes">
+        <dt>
+          <dfn>oncandidatepairadded</dfn> of type <span class="idlAttrType">{{EventHandler}}</span>
+        </dt>
+        <dd>
+          <p>
+            The event type of this event handler is {{candidatepairadded}}.
+          </p>
+          <p>
+            When the [= ICE agent =] has formed a candidate pair, the [= user agent =] MUST queue a task to [= fire an
+            event =] named {{candidatepairadded}} using the {{RTCIceCandidatePairEvent}} interface with the
+            {{RTCIceCandidatePairEvent/candidatePair}} attribute set to the formed candidate pair.
+          </p>
+        </dd>
+        <dt>
+          <dfn>oncandidatepairpruneproposed</dfn> of type <span class="idlAttrType">{{EventHandler}}</span>
+        </dt>
+        <dd>
+          <p>
+            The event type of this event handler is {{candidatepairpruneproposed}}.
+          </p>
+          <p>
+            When the [= ICE agent =] has selected a candidate pair to prune, but before the pruning has actually occurred,
+            the [= user agent =] MUST [= propose a candidate pair pruning =].
+          </p>
+        </dd>
+      </dl>
+    </section>
+    <section>
+      <h2>
+        <dfn>RTCIceCandidatePairEvent</dfn>
+      </h2>
+      <p>
+        The {{RTCIceTransport/candidatepairadded}} and {{RTCIceTransport/candidatepairpruneproposed}} events use the {{RTCIceCandidatePairEvent}} interface.
+      </p>
+      <div>
+        <pre class="idl">[Exposed=Window]
+interface RTCIceCandidatePairEvent : Event {
+  constructor(DOMString type, RTCIceCandidatePairEventInit eventInitDict);
+  readonly attribute RTCIceCandidatePair candidatePair;
```

Comment:
We usually do not introduce dictionary attribute.
We could expose local and remote directly.

Reply:
Another approach discussed during the call: add a RTCIceCandidatePair interface

Reply:
The issue is that an attribute cannot be a dictionary, so it needs to return an interface-version of the candidate pair or contain references to the local and remote directly

Reply:
I agree with making RTCIceCandidate and RTCIceCandidatePair consistent by adding an interface for the latter.

The current dictionary could be changed to an interface like such:
```webidl
interface RTCIceCandidatePair {
    readonly attribute RTCIceCandidate? local;
    readonly attribute RTCIceCandidate? remote;
};
```

Potentially breaking changes due to switching RTCIceCandidatePair from a dictionary to an interface would be
- an ordered map like `{ local: <candidate>, remote: <candidate> }` could no longer be treated implicitly as an RTCIceCandidatePair
- setting `local` or `remote` on an RTCIceCandidatePair instance will fail on the interface

Is there a precedent for a change like this? Or is the suggestion to add a completely new interface by a different identifier?

Reply:
I've switched to local+remote `RTCIceCandidate` for now. We can continue the discussion in the June interim or here.

Reply:
Another option from https://www.w3.org/TR/api-design/#attribute-constant-or-exception-fields - use type `any` for `candidatePair` attribute, and state in the description that `candidatePair` is of type `RTCIceCandidatePair`.